src/LRUCache.jl

```diff
@@ -146,11 +146,14 @@ function Base.getindex(lru::LRU, key)
     end
 end
 function _unsafe_addindex!(lru::LRU{K}, v, key) where K
+    s = lru.by(v)::Int
+    # If entry is larger than entire cache, don't add it
+    s > lru.maxsize && return
     n = LinkedNode{K}(key)
     rotate!(_push!(lru.keyset, n))
-    s = lru.by(v)::Int
     lru.currentsize += s
     lru.dict[key] = (v, n, s)
+    return
```

Comment:
added this so the return type is `nothing` independently of which branch we take. Note the return of `_unsafe_index!` isn't used anywhere in the codebase